src/core/thread/meshcop_dataset_manager.cpp

```diff
@@ -155,7 +155,7 @@ void DatasetManager::HandleTimer(void)
     VerifyOrExit(mLocal.Compare(mNetwork) < 0,);
 
     Register();
-    mTimer.Start(1000);
+    // mTimer.Start(1000);
```

Comment:
The timer is here to ensure that the Leader gets the registration message and guard against message loss.

For the Active Dataset, we should check against the Active Dataset Timestamp in the network's Pending Dataset.  If the Pending Dataset has an Active Dataset Timestamp that is equal or ahead, then we can stop trying to register.  Does that make sense?